apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts

```diff
@@ -81,8 +94,43 @@ router.get(
   '/',
   asyncHandler(async (req, res) => {
     assertCanCreateQuestion(res.locals);
+    if (req.query?.qid) {
+      const qidFull = `__drafts__/${req.query?.qid}`;
+      const prompts = await queryRows(
+        sql.select_ai_question_generation_prompts,
+        { qid: qidFull, course_id: res.locals.course.id.toString() },
+        AiGenerationPromptSchema,
+      );
 
-    res.send(AiGeneratePage({ resLocals: res.locals }));
+      if (prompts && prompts.length > 0) {
+        res.locals.question = await queryRow(
+          sql.select_question_by_qid_and_course,
+          { qid: qidFull, course_id: res.locals.course.id },
+          QuestionSchema,
+        );
+        const variant_id = req.query.variant_id ? IdSchema.parse(req.query.variant_id) : null;
+
+        await getAndRenderVariant(variant_id, null, res.locals);
+        await setQuestionCopyTargets(res);
+        await logPageView('instructorQuestionPreview', req, res);
+        setRendererHeader(res);
+      }
+      const queryUrl = req.originalUrl.split('?')[1]; //only the info after the query seperator
```

Comment:
we keep this parsed out so that we can do htmx requests with query parameters.

Reply:
I think the need for this will go away if we stop using query params to store the QID, and instead make that part of the URL path itself.

Reply:
This was indeed resolved by using the path to specify the question ID.